project-addons/custom_account/voucher_view.xml

```diff
@@ -16,5 +16,16 @@
             </field>
         </record>
 
+        <record id='account_voucher_add_invoice' model='ir.ui.view'>
+            <field name="name">account.voucher.add.invoice</field>
+            <field name="model">account.voucher</field>
+            <field name="inherit_id" ref="account_voucher.view_vendor_payment_form"/>
+            <field name="arch" type="xml">
+                <xpath expr="/form/sheet/notebook/page/field[@name='line_dr_ids']/tree/field[@name='account_id']" position="after">
```

Comment:
Podéis hacerlo de forma relativa, que siempre es más seguro, por si algún móudlo modifica la estructura.
Es decir:
expr="//field[@name='line_dr_ids']/tree/field[@name='account_id']"